### PR TITLE
Split torch.nn docs

### PR DESCRIPTION
**Pull request by jlin27**
Related Issue: https://github.com/pytorch/pytorch/issues/20984
Addresses the comments in this PR: https://github.com/pytorch/pytorch/pull/34923
- Only focuses on breaking up nn.rst into separate files. No longer contains doc reorganization changes
- Removes nn/index.rst and moves contents of that file under nn.rst
- Corrected the repeated docs issue (removed unneeded sections from nn_pooling_layers.rst)
- Corrected the space the file name - "nn_normalization_layers" and ".rst"

### CODE REVIEW

File: docs/source/nn.rst [hunk 1]
```diff
@@ -1,1001 +1,25 @@
...
-
-Quantized Functions
---------------------
-
-Quantization refers to techniques for performing computations and storing tensors at lower bitwidths than
-floating point precision. PyTorch supports both per tensor and per channel asymmetric linear quantization. To learn more how to use quantized functions in PyTorch, please refer to the :ref:`quantization-doc` documentation.
+.. toctree::
+   :maxdepth: 1
+
+   nn/nn_containers
```
Comment (zou3519) [inline docs/source/nn.rst:8]: I'm not sure these should be alphabetized: it's important that parameters comes first and a lot of these were grouped based on relevancy (for example, convolution, padding, and pooling all relate to image processing, so they were adjacent to each other)

**Review by zou3519 — CHANGES_REQUESTED** [review]
I don't think it makes sense to alphabetize the TOC -- previously the ordering was such that relevant sections are grouped together

**Comment by zou3519** [comment]
@vadimkantorov had a great point in https://github.com/pytorch/pytorch/pull/34923#issuecomment-603569284, old links like https://pytorch.org/docs/stable/nn.html#torch.nn.ConvTranspose1d will fail to work . Keeping around a "shadow nn page" like in the original PR would preserve those links, but has some downsides:
- cross-referencing nn documentation links from other doc pages or searching might break if there are multiple doc entries of the same function
- we should make a decision on if developers are updating the docs, if they need to update both the shadow page and these new pages or if they're only going to update the new pages.

It would nice if the old links magically redirected correctly, but I imagine that could be difficult to implement. 

Some other alternatives I can think of:
- since we own the forums, we could just go in and change every link to their corresponding new link

**Comment by vadimkantorov** [comment]
@zou3519 one hack for keeping links work would be to have some JavaScript client-side URL redirector in `nn.html`

File: docs/source/nn.rst (hunk 1 above)
Comment (jlin27) [reply docs/source/nn.rst]: Okay that makes sense to me. I just pushed new changes that revers the alphabetization

**Comment by jlin27** [comment]
Re: adding redirects - I tested adding these <meta> tags in a fork of the pytorch/serve repo and successfully go the redirects to work: https://github.com/jlin27/serve/blob/gh-pages/inference_api.html#L4-L8 ([Source](https://dev.to/steveblue/setup-a-redirect-on-github-pages-1ok7)). So https://jlin27.github.io/serve/inference_api.html redirects to https://example.com/. 

This example was for a gh-pages setup; however, so the serving mechanisms for pytorch/pytorch is a bit different but I think the tags should still work the same way. I can try testing by adding the redirect tags to master and push to the site and see if that works. Thoughts?

**Comment by vadimkantorov** [comment]
I think the difficulty is that depending on the function in `/nn.html#somethashtag` the redirect needs to go to a different page. So a `<meta>` tag won't suffice and dynamic JavaScript for parsing this may be needed
